examples/snmp/pfsense.js

```diff
@@ -0,0 +1,889 @@
+			var new_value = result[_this.oid];
+			var time_diff = (new_time - old_time) / 1000;
+			_this.result = (new_value - old_value) / time_diff;
+			next(_this);
+		});
+	}, 5000);
+}
+
+function divide_by_cpu_count(last, next) {
+	var nCPU = get_result('1.3.6.1.2.1.25.3.3.1.2.cpus');
+	this.result = this.result / nCPU;
+	next(this);
+}
+
+function multiply_result(number) {
+	return function (last, next) {
+		this.result = this.result * number;
+		next(this);
+	};
+}
+
+
+
+
+function preprocess_params(cb) {
+	var arrayFn = [];
+	snmp_parameters.forEach(function (p) {
+		if (p.walk) {
+			arrayFn.push(function (callback) {
+				createSNMPSession().walk(p.walk.ref, function (result, error) {
+					p.walk.callback(p, result);
+					callback(p);
+				});
+			});
+
+		}
+	});
+
+	execute_all(arrayFn, function (result) {
+		cb();
+	});
+}
+
+
+function generate_variable(p, result, first_execution_date, callback) {
+	if (p.unit) {
+		p.label += ' (' + p.unit + ')';
+	}
+	if(result[p.oid]){
+		p.result = result[p.oid];
+	}
+	if (p.convert && typeof (p.convert) == 'function') {
+		p.result = p.convert();
+	}
+	p.uid = p.oid;
+	if (!p.uid) {
+		p.uid = p.walk.ref + '.' + p.walk.name;
+	}
+	p.first_execution_date = first_execution_date;
+	if (p.postProcess && p.postProcess.length) {
+		console.log('Postprocessing ' + p.label);
+		return execute_seq.apply(p, [p.postProcess, function (result) {
+			console.log('Postprocessing done ' + p.label);
+			callback(p);
+		}]);
+	}
+	callback(p);
+}
+
+function execute_get(param) {
+	return function (callback) {
+		// console.log("excuting for parameter: " + param.label)
+		function run() {
+			if (param.oid && param.snmpget !== false) {
+				createSNMPSession().get([param.oid], function (result, error) {
+					if (error) {
+						console.error(error);
+						D.failure(D.errorType.GENERIC_ERROR);
+					}
+
+					generate_variable(param, result, new Date(), function (variable) {
+						callback(variable);
+					});
+
+				});
+			} else {
+				generate_variable(param, {}, new Date(), function (variable) {
+					callback(variable);
+				});
+			}
+		}
+		if (param.preProcess && param.preProcess.length) {
+			console.log('Preprocessing ' + param.label);
+			return execute_seq.apply(param, [param.preProcess, function () {
+				console.log('Preprocessing done ' + param.label);
+				run();
+			}]);
+		} else {
+			run();
+		}
+
+	};
+}
+
+
+
+/**
+* @remote_procedure
+* @label Validate Association
+* @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
+*/
+function validate() {
+	createSNMPSession().get(['1.3.6.1.4.1.12325.1.200.1.2.1.0'], function (result, error) {
+		if (error) {
+			console.error(error);
+			D.failure(D.errorType.GENERIC_ERROR);
+		}
+		D.success();
+	});
+
+}
+
+
+/**
+* @remote_procedure
+* @label Get Device Variables
+* @documentation This procedure is used for retrieving device * variables data
```

Comment:
Maybe a bit more detailed documentation text would be better (same for the label)